azurerm/internal/services/storage/resource_arm_storage_account.go

```diff
@@ -1239,20 +1239,25 @@ func resourceArmStorageAccountRead(d *schema.ResourceData, meta interface{}) err
 		}
 	}
 
-	queueClient, err := storageClient.QueuesClient(ctx, *account)
-	if err != nil {
-		return fmt.Errorf("Error building Queues Client: %s", err)
-	}
+	// queue is only available for certain tier and kind (as specified below)
+	if resp.Sku.Tier == storage.Standard {
```

Comment:
resp.Sku is a nilable object here, so we'd need to nil-check this first:

```suggestion
	if resp.Sku == nil {
		return fmt.Errorf("Error retrieving Storage Account %q (Resource Group %q): `sku` was nil", name, resourceGrup)
	}
	
	if resp.Sku.Tier == storage.Standard {
```

Reply:
Good to know 😉